### PR TITLE
Pretty-print the features produced by buildFeatures 

### LINKED ISSUE #163: Pretty-print the features produced by buildFeatures and submit job
Provide a pretty-print flag in buildFeautre(https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/client.py#L213). If it's true, then print the features that are constructed by buildFeature in pretty-print form.

Same for get_offline_feature: https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/client.py#L392 and https://github.com/linkedin/feathr/blob/main/feathr_project/feathr/client.py#L514.

### PR DESCRIPTION
**Pull request by ahlag**
## Description
- Added pretty print flag to build_features

Resolves #163 

## How was this patch tested?
Ran `pytest -s feathr_project/test/test_feature_materialization.py -k "test_build_feature_verbose"` to check the pretty-printed features

## Does this PR introduce any user-facing changes?
Introduces `verbose` argument to pretty print features

### CODE REVIEW

File: feathr_project/feathr/client.py
Comment (hangfei) [inline feathr_project/feathr/client.py:9]: this seems wrong wrong?

**Comment by hangfei** [comment]
For pretty print, let's use a util so the logics can be unified and reused.

I am thinking about this format:

These are the features you are going to register/build/materialize:
feature1, feature2, and feature3.

File: feathr_project/feathr/client.py
Comment (ahlag) [reply feathr_project/feathr/client.py]: Oops sorry. Removed `from xmlrpc.client import Boolean`

**Comment by ahlag** [comment]
@hangfei 
Okay sure.

**Comment by hangfei** [comment]
> 

may  not be critical. just re-ran the tests to see if it passes

**Comment by hangfei** [comment]
For util, let's create a new py file in the same direcotry.

**Comment by hangfei** [comment]
As long as your local test that you added passes, it should be fine. There is a crendential issue on a forked PR(we are fixing it and hopefully it can be fixed in 1-2 weeks).

Since you are just adding some pretty-print, i don't think it will break the e2e tests.

I will ran the e2e tests after it's merged.

File: feathr_project/feathr/client.py
Comment (hangfei) [reply feathr_project/feathr/client.py:246]: verbose sounds good. The end goal is to unify them to print out some execution plans for users to review/proofread to catch early errors or debug.

**Comment by ahlag** [comment]
@hangfei 
This PR is up for review.

I tested the verbose flag with the following command:
```
pytest -s feathr_project/test/test_feature_materialization.py -k "test_build_feature_verbose"                                                                         ─╯
=========================================================================== test session starts ===========================================================================
platform darwin -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/changyonglik/Desktop/opensource/feathr/feathr_project
collected 5 items / 4 deselected / 1 selected

feathr_project/test/test_feature_materialization.py 2022-05-08 14:15:15.225 | INFO     | feathr._envvariableutil:get_environment_variable:64 - DATABRICKS_WORKSPACE_TOKEN_VALUE is not set in the environment variables.
2022-05-08 14:15:15.233 | INFO     | feathr._envvariableutil:get_environment_variable:64 - REDIS_PASSWORD is not set in the environment variables.
("request_features is the achor feature of ['trip_distance', "
 "'f_is_long_trip_distance', 'f_day_of_week']>")
"user_embemdding_derived is the derived feature of ['user_embedding']>"
.
```

File: feathr_project/test/test_feature_materialization.py
Comment (hangfei) [inline feathr_project/test/test_feature_materialization.py:76]: what's this test for? Is this related to this PR? I am a bit out of context for this change.

File: feathr_project/feathr/client.py
Comment (hangfei) [inline feathr_project/feathr/client.py:451]: we should only print feature_query. Feature anchor is a super set of feature_query. 

For example, i have an anchor and it contains f1, f2, f3.

But I only want f1 and f2. Then my feature_query = f1, f2.

Giving out whole anchors is confusing.

File: feathr_project/test/test_feature_materialization.py
Comment (ahlag) [reply feathr_project/test/test_feature_materialization.py:76]: Oops, sorry. My mistake. Copy-paste error.

File: feathr_project/feathr/client.py
Comment (ahlag) [reply feathr_project/feathr/client.py:451]: @hangfei  
Ok no problem. I can build a PrettyPrinter for `feature_query`. Just one clarification. (Really sorry for the questions)
1. Is `feature_query` supposed to be printed only in `get_offline_features` or should it be printed in `materialize_features` and `build_features` as well?
1. If `feature_query` needs to be printed in `materialize_features` and `build_features`, there is no `feature_query` argument in `materialize_features` and `build_features`. How can `feature_query` be referred in `materialize_features` and `build_features`?
https://github.com/linkedin/feathr/blob/7dd36a5496c301006136a0dde5caf07306d26464/feathr_project/feathr/client.py#L213
https://github.com/linkedin/feathr/blob/7dd36a5496c301006136a0dde5caf07306d26464/feathr_project/feathr/client.py#L530

**Comment by ahlag** [comment]
@hangfei 
I have finished implementing the pretty-prints. Could you take a look? `test_get_offline_features_verbose` and `test_materialize_features_verbose` fail because it requires REDIS and KAFKA settings. The pretty-print works without a problem.

```
❯ pytest -s feathr_project/test/test_feature_materialization.py -k "test_build_feature_verbose"                                                               
====================================================================== test session starts =======================================================================
platform darwin -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/changyonglik/Desktop/opensource/feathr/feathr_project
collected 7 items / 6 deselected / 1 selected

feathr_project/test/test_feature_materialization.py 2022-05-17 00:00:06.106 | INFO     | feathr._envvariableutil:get_environment_variable:66 - DATABRICKS_WORKSPACE_TOKEN_VALUE is not set in the environment variables, fetching the value from Key Vault
2022-05-17 00:00:06.115 | INFO     | feathr._envvariableutil:get_environment_variable:66 - REDIS_PASSWORD is not set in the environment variables, fetching the value from Key Vault
("request_features is the achor of ['trip_distance', "
 "'f_is_long_trip_distance', 'f_day_of_week']")
```

```
❯ pytest -s feathr_project/test/test_feature_materialization.py -k "test_get_offline_features_verbose"                                                         
====================================================================== test session starts =======================================================================
platform darwin -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/changyonglik/Desktop/opensource/feathr/feathr_project
collected 7 items / 6 deselected / 1 selected

feathr_project/test/test_feature_materialization.py 2022-05-16 23:59:32.888 | INFO     | feathr._envvariableutil:get_environment_variable:66 - DATABRICKS_WORKSPACE_TOKEN_VALUE is not set in the environment variables, fetching the value from Key Vault
2022-05-16 23:59:32.896 | INFO     | feathr._envvariableutil:get_environment_variable:66 - REDIS_PASSWORD is not set in the environment variables, fetching the value from Key Vault
Features in feature_query: ['f_location_avg_fare']
```

```
❯ pytest -s feathr_project/test/test_feature_materialization.py -k "test_materialize_features_verbose"                                                                                           
======================================================================================= test session starts ========================================================================================
platform darwin -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
rootdir: /Users/changyonglik/Desktop/opensource/feathr/feathr_project
collected 7 items / 6 deselected / 1 selected

feathr_project/test/test_feature_materialization.py 2022-05-17 00:13:00.655 | INFO     | feathr._envvariableutil:get_environment_variable:66 - DATABRICKS_WORKSPACE_TOKEN_VALUE is not set in the environment variables, fetching the value from Key Vault
2022-05-17 00:13:00.664 | INFO     | feathr._envvariableutil:get_environment_variable:66 - REDIS_PASSWORD is not set in the environment variables, fetching the value from Key Vault
Yes
Materialization features in settings: ['f_location_avg_fare', 'f_location_max_fare']
2022-05-17 00:13:00.746 | INFO     | feathr._envvariableutil:get_environment_variable:66 - KAFKA_SASL_JAAS_CONFIG is not set in the environment variables, fetching the value from Key Vault
```

**Comment by hangfei** [comment]
This looks great! Very impressive test coverage!

**Review by hangfei — APPROVED** [review]

**Comment by xiaoyongzhu** [comment]
Thanks for the contribution @ahlag !

**Comment by chinmaytredence** [comment]
@ahlag Should we put the utils.py inside the feathr directory

**Comment by ahlag** [comment]
@chinmaytredence (cc: @hangfei )
Will the pretty-prints be used elsewhere? (Just confirming)

**Comment by chinmaytredence** [comment]
@ahlag this is breaking the uvicorn run when we add this feathr project into requirements.txt
Screenshot 2022-05-18 at 9 16 17 AM
t

**Comment by ahlag** [comment]
@chinmaytredence 
Could you share the steps you used to reproduce this? I will take a look into this.

**Comment by chinmaytredence** [comment]
@ahlag you can start a flask/fastapi project and add this "git+https://github.com/linkedin/feathr.git@main#subdirectory=feathr_project"
into the requirements.txt to use this as a library. 
If the utils.py is not inside the "fethr" directory then it will not be downloaded into the dependencies. Thus it breaks the running of application.

**Comment by chinmaytredence** [comment]
Screenshot 2022-05-18 at 9 22 17 AM
Screenshot 2022-05-18 at 9 22 50 AM

**Comment by ahlag** [comment]
@chinmaytredence 
Ok. I will create a quick fix and then tag the maintainers and you for a review. Thank you!

**Comment by ahlag** [comment]
@chinmaytredence @hangfei 
I have created a fix here. Could you review it? I have tested it locally.
https://github.com/linkedin/feathr/pull/273

**Comment by chinmaytredence** [comment]
@ahlag validated it. Working now. Thanks for the fix.

**Pull request merged**
